src/app/qgslayertreeviewfilterindicator.cpp

```diff
@@ -37,8 +37,12 @@ void QgsLayerTreeViewFilterIndicatorProvider::onIndicatorClicked( const QModelIn
   if ( !QgsLayerTree::isLayer( node ) )
     return;
 
-  QgisApp::instance()->layerSubsetString( QgsLayerTree::toLayer( node )->layer() );
+  QgsMapLayer *layer = QgsLayerTree::toLayer( node )->layer();
+  QgsVectorLayer *vl = qobject_cast<QgsVectorLayer *>( layer );
+  if ( vl && vl->isEditable() )
+    return;
```

Comment:
Instead of silently returning (which gives users no idea why this operation doesn't work when it usually does), could we show an informative message in the message bar?

Reply:
Done